[Bug] Build Error: `numeric_limits` is not a member of `std`.

### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.


### Ray Component

Ray Core

### What happened + What you expected to happen

1. Bug:
```
external/com_github_jupp0r_prometheus_cpp/core/src/histogram.cc: In member function ‘prometheus::ClientMetric prometheus::Histogram::Collect() const’:
external/com_github_jupp0r_prometheus_cpp/core/src/histogram.cc:54:38: error: ‘numeric_limits’ is not a member of ‘std’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                      ^~~~~~~~~~~~~~
external/com_github_jupp0r_prometheus_cpp/core/src/histogram.cc:54:53: error: expected primary-expression before ‘double’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                                     ^~~~~~
external/com_github_jupp0r_prometheus_cpp/core/src/histogram.cc:54:53: error: expected ‘:’ before ‘double’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                                     ^~~~~~
      |                                                     :
external/com_github_jupp0r_prometheus_cpp/core/src/histogram.cc:54:53: error: expected primary-expression before ‘double’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                                     ^~~~~~
external/com_github_jupp0r_prometheus_cpp/core/src/histogram.cc:54:53: error: expected ‘)’ before ‘double’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                                     ^~~~~~
      |                                                     )
```
2. Expected behavior: build successfully
3. sysinfo
OS: ArchLinux
GCC: 11.1.0
Bazel: 4.2.0
Java: 11.0.12+7

### Reproduction script

N/A build error

### Anything else

N/A

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

--- a/thirdparty/patches/prometheus-windows-headers.patch
+++ b/thirdparty/patches/prometheus-windows-headers.patch
@@ -1,7 +1,8 @@
 diff --git core/src/histogram.cc core/src/histogram.cc
 --- core/src/histogram.cc
 +++ core/src/histogram.cc
-@@ -6,1 +6,2 @@
+@@ -6,1 +6,3 @@
  #include <numeric>
 +#include <stdexcept>
++#include <limits>
 -- 

